pyproject.toml

```diff
@@ -53,6 +53,62 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+alignment = [
+    "dtw-python",
+    "numba",
+]
+annotation = [
+    "attrs",
+    "hmmlearn",
+    "numba",
+    "pyod",
+]
+classification = [
+    "esig",
+    "mrsqm",
+    "numba",
+    "tensorflow",
+    "tsfresh",
+]
+clustering = [
+    "numba",
+    "tslearn",
+]
+forecasting = [
+    "hcrystalball",
+    "pmdarima",
+    "prophet",
+    "scikit-optimize",
+    "statsforecast",
+    "statsmodels",
+    "tbats",
+]
+networks = [
+    "tensorflow",
+]
+param_est = [
+    "seasonal",
+    "statsmodels",
+]
+proba = [
+    "tensorflow-probability",
```

Comment:
this is no longer needed, actually it's not recommended anymore.

Reply:
Okay, are you suggesting extra removal, or removal even from `all_extras`? Won't it break `proba` component? 

(Again, I don'tuse it so no idea.)

Reply:
hm, it shouldn't break anything since (a) it's a soft dependency, (b) it's not imported anywhere except the probability distributions in `proba`.

I think we still need to deprecate it, as someone might be using it though - but for now we don't need to add an extra `proba` dep set imo.